python/mrtrix3/commands/dwimotioncorrect.py

```diff
@@ -0,0 +1,333 @@
+    # options
+    options = cmdline.add_argument_group('Options for the dwimotioncorrect script')
+    options.add_argument('-mask', type=app.Parser.ImageIn(), help='Manually provide a mask image for motion correction')
+    options.add_argument('-lmax', help='SH basis order per shell (default = maximum with at least 30% oversampling)')
+    options.add_argument('-rlmax', help='Reduced basis order per component for registration (default = 4,2,0 or lower if needed)')
+    options.add_argument('-reg', help='Regularization for dwirecon (overrides config; default = 0.005)')
+    options.add_argument('-zreg', help='Regularization for dwirecon (overrides config; default = 0.001)')
+    options.add_argument('-setup', type=app.Parser.FileIn(), help='Configuration setup file (json structured)')
+    options.add_argument('-mb', help='Multiband factor (default = 1)')
+    options.add_argument('-sorder', help='Slice order (default = 2,1, for odd-even)')
+    options.add_argument('-sspwidth', help='Slice thickness for Gaussian SSP (default = 1)')
+    options.add_argument('-sspfile', type=app.Parser.FileIn(), help='Slice sensitivity profile as vector')
+    options.add_argument('-fieldmap', type=app.Parser.ImageIn(), help='B0 field map for distortion correction')
+    options.add_argument('-fieldidx', help='Index of volume to which field map is aligned (default = 0)')
+    options.add_argument('-pe_table', type=app.Parser.FileIn(), help='Phase encoding table in MRtrix format')
+    options.add_argument('-pe_eddy', nargs=2, type=app.Parser.FileIn(), help='Phase encoding table in FSL acqp/index format')
+    options.add_argument('-priorweights', type=app.Parser.FileIn(), help='Import prior slice weights')
+    options.add_argument('-fixedweights', type=app.Parser.FileIn(), help='Import fixed slice weights')
+    options.add_argument('-fixedmotion', type=app.Parser.FileIn(), help='Import fixed motion traces')
+    options.add_argument('-driftfilter', action='store_true', help='Filter motion for drift')
+    options.add_argument('-voxelweights', type=app.Parser.ImageIn(), help='Import fixed voxel weights')
+    options.add_argument('-export_motion', type=app.Parser.FileOut(), help='Export rigid motion parameters')
+    options.add_argument('-export_weights', type=app.Parser.FileOut(), help='Export slice weights')
+    app.add_dwgrad_import_options(cmdline)
+
+
+
+def execute(): #pylint: disable=unused-variable
+    try:
+        import scipy
+    except ImportError:
+        raise MRtrixError('SciPy required: https://www.scipy.org/install.html')
+    # import metadata
+    grad_import_option = app.dwgrad_import_options()
+    pe_import_option = []
+    if app.ARGS.pe_table:
+        pe_import_option = ['-import_pe_table', app.ARGS.pe_table]
+    elif app.ARGS.pe_eddy:
+        pe_import_option = ['-import_pe_eddy', app.ARGS.pe_eddy[0], app.ARGS.pe_eddy[1]]
+    # check output path - code no longer works; now done internally??
```

Comment:
Correct: arguments of type `(Directory|File|Image)Out` are now automatically checked, so devs are no longer responsible for adding these function calls.
https://github.com/MRtrix3/mrtrix3/blob/1f9d3b485e0e3505a89848fc5e7747b7f50d350c/python/mrtrix3/app.py#L181-L192